packages/block-library/src/columns/transforms.js

```diff
@@ -8,8 +8,55 @@ import {
 
 const MAXIMUM_SELECTED_BLOCKS = 6;
 
+function fromMediaText( attributes, innerBlocks ) {
+	const {
+		align,
+		backgroundColor,
+		mediaAlt: alt,
+		mediaId: id,
+		mediaPosition,
+		mediaSizeSlug: sizeSlug,
+		mediaType,
+		mediaUrl: url,
+		mediaWidth,
+		verticalAlignment,
+	} = attributes;
+	let media;
+	if ( mediaType === 'image' || ! mediaType ) {
+		const imageAttrs = { id, alt, url, sizeSlug };
+		const linkAttrs = {
+			href: attributes.href,
+			linkClass: attributes.linkClass,
+			linkDestination: attributes.linkDestination,
+			linkTarget: attributes.linkTarget,
+			rel: attributes.rel,
+		};
+		media = [ 'core/image', { ...imageAttrs, ...linkAttrs } ];
+	} else {
+		media = [ 'core/video', { id, src: url } ];
+	}
+	const innerBlocksTemplate = [
+		[ 'core/column', { width: `${ mediaWidth }%` }, [ media ] ],
+		[ 'core/column', { width: `${ 100 - mediaWidth }%` }, innerBlocks ],
+	];
+	if ( mediaPosition === 'right' ) {
+		innerBlocksTemplate.reverse();
+	}
+	return createBlock(
+		'core/columns',
+		{ align, backgroundColor, verticalAlignment },
+		createBlocksFromInnerBlocksTemplate( innerBlocksTemplate )
+	);
+}
+
 const transforms = {
 	from: [
+		{
+			type: 'block',
+			blocks: [ 'core/media-text' ],
+			isMatch: ( { mediaId } ) => !! mediaId,
```

Comment:
This was intended to not offer the transform when no media has been selected. I couldn't get that to work (even just returning `false` here) and have left it here to inquire about it. Any ideas @ntsekouras? We could remove it. As of now, I made the transform default to inserting an empty Image block when transforming before any media has been selected.

Reply:
That's a tricky one :).  The problem here is that `isMatch` is checked in a different flow than the actual transformation. It is checked in `getPossibleBlockTransformations` that is used to return the available block types to show in `Block Switcher` list.

After that check that has found `Columns` to be an eligible match (with 2 transformations though :) ) the actual transformation selection happens in a different place here: https://github.com/WordPress/gutenberg/blob/5cd4f3f76ba80049a4b45ccba9f5f023804730b2/packages/blocks/src/api/factory.js#L441

which selects the first match which happens to be the first declared one. You can check that if you put your new transform below the existing one, it will always choose the old one.

The wildcard `*` complicates things on how to combine other transforms. Recently I looked into a related problem here: https://github.com/WordPress/gutenberg/issues/29296, with no clear path forwards for now.

Reply:
That's a tricky one :).  The problem here is that `isMatch` is checked in a different flow than the actual transformation. It is checked in `getPossibleBlockTransformations` that is used to return the available block types to show in `Block Switcher` list.

After that check that has found `Columns` to be an eligible match (with 2 transformations though :) ) the actual transformation selection happens in a different place here: https://github.com/WordPress/gutenberg/blob/5cd4f3f76ba80049a4b45ccba9f5f023804730b2/packages/blocks/src/api/factory.js#L441

which selects the first match which happens to be the first declared one. You can check that if you put your new transform below the existing one, it will always choose the old one.

The wildcard `*` complicates things on how to combine other transforms. Recently I looked into a related problem here: https://github.com/WordPress/gutenberg/issues/29296, with no clear path forwards for now.

**--edit**

After digging a bit more it seems that the prioritizations of transforms have been implemented a long ago here: https://github.com/WordPress/gutenberg/pull/5701

So by adding a `priority: 1` prop in your `transform` will return the wanted transformation.

Reply:
>You can check that if you put your new transform below the existing one, it will always choose the old one.

True, I had noticed that as I started this with the new transform below the existing. I've pushed another commit to put it back below and use `priority`. I still couldn't get `isMatch` to prevent the transform but it's unnecessary for this PR as I think it is always preferable to use this transform even if no media has been selected.

Thanks for looking into this!